server/service/integration_mdm_test.go

```diff
@@ -587,6 +589,179 @@ func (s *integrationMDMTestSuite) TestMDMAppleUnenroll() {
 	s.Do("PATCH", fmt.Sprintf("/api/latest/fleet/mdm/hosts/%d/unenroll", h.ID), nil, http.StatusOK)
 }
 
+func (s *integrationMDMTestSuite) TestMDMAppleGetEncryptionKey() {
+	t := s.T()
+	ctx := context.Background()
+
+	// create a host
+	host, err := s.ds.NewHost(context.Background(), &fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		PolicyUpdatedAt: time.Now(),
+		SeenTime:        time.Now().Add(-1 * time.Minute),
+		OsqueryHostID:   ptr.String(t.Name()),
+		NodeKey:         ptr.String(t.Name()),
+		UUID:            uuid.New().String(),
+		Hostname:        fmt.Sprintf("%sfoo.local", t.Name()),
+		Platform:        "darwin",
+	})
+	require.NoError(t, err)
+
+	// add an encryption key for the host
+	cert, _, _, err := s.fleetCfg.MDM.AppleSCEP()
+	require.NoError(t, err)
+	parsed, err := x509.ParseCertificate(cert.Certificate[0])
+	require.NoError(t, err)
+	recoveryKey := "AAA-BBB-CCC"
+	encryptedKey, err := pkcs7.Encrypt([]byte(recoveryKey), []*x509.Certificate{parsed})
+	require.NoError(t, err)
+	base64EncryptedKey := base64.StdEncoding.EncodeToString(encryptedKey)
+
+	err = s.ds.SetOrUpdateHostDiskEncryptionKey(ctx, host.ID, base64EncryptedKey)
+	require.NoError(t, err)
+
+	// request with no token
+	res := s.DoRawNoAuth("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusUnauthorized)
+	res.Body.Close()
+
+	// encryption key not procesed yet
+	resp := getHostEncryptionKeyResponse{}
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusNotFound, &resp)
+
+	// unable to decrypt encryption key
+	err = s.ds.SetHostDiskEncryptionKeyStatus(ctx, []uint{host.ID}, false)
+	require.NoError(t, err)
+	resp = getHostEncryptionKeyResponse{}
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusNotFound, &resp)
+
+	// no activities created so far
+	activities := listActivitiesResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &activities)
+	found := false
+	for _, activity := range activities.Activities {
+		if activity.Type == "read_host_disk_encryption_key" {
+			found = true
+		}
+	}
+	require.False(t, found)
+
+	// decryptable key
+	checkDecryptableKey := func(u fleet.User) {
+		err = s.ds.SetHostDiskEncryptionKeyStatus(ctx, []uint{host.ID}, true)
+		require.NoError(t, err)
+		resp = getHostEncryptionKeyResponse{}
+		s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusOK, &resp)
+		require.Equal(t, recoveryKey, resp.EncryptionKey.DecryptedValue)
+
+		// use the admin token to get the activities
+		currToken := s.token
+		defer func() { s.token = currToken }()
+		s.token = s.getTestAdminToken()
+		activities = listActivitiesResponse{}
+		s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &activities, "order_key", "created_at")
```

Comment:
You probably want order_direction descending. I think I created a helper func to validate the latest created activity, that might be helpful here (`lastActivityMatches`, it's in the enterprise integration tests but it can probably be detached from any specific integration test by using an interface or `*withServer` instead).